dbt_subprojects/solana/models/_sector/staking/solana/staking_solana_validator_stake_account_epochs_raw.sql

```diff
@@ -0,0 +1,40 @@
+{{ config(
+    schema = 'staking_solana'
+    , alias = 'validator_stake_account_epochs_raw'
+    , materialized = 'incremental'
+    , file_format = 'delta'
+    , incremental_strategy = 'merge'
+    , unique_key = ['epoch', 'epoch_time', 'stake_account', 'vote_account']
+    , incremental_predicates = [incremental_predicate('DBT_INTERNAL_DEST.epoch_time')]
+)
+}}
+
+--don't expose this table in DE
+with 
+    base as (
+        SELECT 
+            epoch.epoch
+            , epoch.block_time as epoch_time
+            , epoch.epoch_start_slot
+            , epoch.epoch_next_start_slot
+            , vote.stake_account
+            , max_by(vote.vote_account, vote.block_time) as vote_account --latest delegated vote before epoch started. then get the rewards from accounts per epoch.
+        FROM {{ ref('staking_solana_stake_account_delegations') }} vote
```

Comment:
the table left joined is filtered incrementally, should this one be too?

Reply:
no, the base table needs full history since we don't know the last delegation changed date